keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -116,6 +117,43 @@ class ClusterHandler(
       .getActiveServerGroups(resource)
       .byRegion()
 
+  override suspend fun willTakeAction(
+    resource: Resource<ClusterSpec>,
+    resourceDiff: ResourceDiff<Map<String, ServerGroup>>
+  ): ActionDecision {
+    // we can't take any action if there is more than one active server group
+    //  AND the current active server group is unhealthy
+    val potentialInactionableRegions = mutableListOf<String>()
+    val inactionableRegions = mutableListOf<String>()
+    resourceDiff.toIndividualDiffs().forEach { diff ->
+      if (diff.hasChanges() && diff.isEnabledOnly()) {
+        potentialInactionableRegions.add(diff.desired.location.region)
+      }
+    }
+    if (potentialInactionableRegions.isNotEmpty()) {
+      val activeServerGroups = cloudDriverService.getActiveServerGroups(resource)
+      activeServerGroups.forEach { serverGroup ->
+        val healthy = serverGroup.instanceCounts?.isHealthy(
+          resource.spec.deployWith.health,
+          resource.spec.resolveCapacity(serverGroup.location.region)
+        ) == true
+        if (!healthy && potentialInactionableRegions.contains(serverGroup.location.region)) {
+          inactionableRegions.add(serverGroup.location.region)
+        }
+      }
+      if (inactionableRegions.isNotEmpty()) {
+        return ActionDecision(
+          willAct = false,
+          message = "There is more than one server group enabled " +
+            "but the latest is not healthy in ${inactionableRegions.joinToString(" and ")}. " +
```

Comment:
This will look a bit weird if you have 3 regions, no? Maybe just use a comma instead?

Reply:
It will look a bit weird, but I think ok for now (I just did some handling of this in the unhealthy notification which I can add if people think it's super weird).

It'll look like 
> `not healthy in us-east-1 and us-west-2 and eu-west-1`
Vs what we probably want is 
> `not healthy in us-east-1, us-west-2, and eu-west-1`
or
> `not healthy in all regions`

I'm going to leave this for a follow up PR because that change is small